### PR TITLE
Support screenshot outside viewport

### PR DESCRIPTION
**Pull request by soundofspace**
The end goal of this pull request is to use the flag `captureBeyondViewport` added in chrome 87, and migrate all screenshot logic to this approach.
This would solve #13, #21, and in general make the screenshot logic much simpler.

This initial commit is in no way meant to be merged, but is here to show that is now possible using the default/latest chrome in the unblocked repo.

What still needs to be done:
- refactor code
- test if scale still works as expected
- extend tests, and remove saving images to /tmp
- test in which versions of chrome this works (find oldest working version)

These are the images created in tests, using the old and new logic (left and right respectively):
Should be able to take a full page screenshot
full-merged

Should screenshot only the visible page
visible-merged

Should be able to take a viewport screenshot:
viewport-merged

Should be able to take a clipped rect screenshot
clipped-merged

New test, should be able to take a clipped rect screenshot outside of viewport
clipped-outside-new

### CODE REVIEW

**Comment by blakebyrnes** [comment]
Awesome. Thanks! Probably worth creating the tmp directory as `os.tmpdir()/screenshots` and creating it at the beginning of the Page tests. Then we can see if this works in the last few versions and Chrome 97.

File: agent/core/lib/Page.ts [hunk 1]
```diff
@@ -444,53 +444,19 @@ export default class Page extends TypedEventEmitter<IPageLevelEvents> implements
...
-        };
-      }
-    } else {
-      const viewportRect = clipRect
-        ? this.trimClipToSize(clipRect, viewportSize)
-        : { x: 0, y: 0, ...viewportSize };
-      clip = {
-        x: pageX + viewportRect.x,
-        y: pageY + viewportRect.y,
-        width: Math.floor(viewportRect.width / scale),
```
Comment (blakebyrnes) [inline agent/core/lib/Page.ts:476]: Did you verify this scale adjustment is not needed?

File: agent/core/test/Page.test.ts [hunk 2]
```diff
@@ -201,6 +207,7 @@ describe('Pages', () => {
          <div style="height: 200px; width: 100px; background:red">&nbsp;</div>`,
       );
       const screenshot = await page.screenshot({ format: 'png' });
+      writeFileSync(`${screenshotDir}/1-viewport.png`, screenshot);
```
Comment (blakebyrnes) [inline agent/core/test/Page.test.ts:210]: Do you think there's any reason to keep these assets? I understood the purpose when we were testing, but I'm not sure they add much value as test outputs. Open to something I'm missing though!

Comment (soundofspace) [reply agent/core/test/Page.test.ts:210]: Not really, could help if they break it again, mainly to see if the content is blank or as expected. But I agree this clutters the tests, and if needed can easily be added again. Will remove them.

**Pull request merged**